lib/usagereporter/usagereporter.go

```diff
@@ -367,24 +391,28 @@ func NewUsageReporter[T any](options *Options[T]) *UsageReporter[T] {
 		options.SubmitClock = clockwork.NewRealClock()
 	}
 
-	return &UsageReporter[T]{
+	reporter := &UsageReporter[T]{
 		Entry: options.Log.WithField(
 			trace.Component,
 			teleport.Component(teleport.ComponentUsageReporting),
 		),
-		events:                make(chan []*SubmittedEvent[T], 1),
-		submissionQueue:       make(chan []*SubmittedEvent[T], 1),
-		submit:                options.Submit,
-		clock:                 options.Clock,
-		submitClock:           options.SubmitClock,
-		minBatchSize:          options.MinBatchSize,
-		maxBatchSize:          options.MaxBatchSize,
-		maxBatchAge:           options.MaxBatchAge,
-		maxBufferSize:         options.MaxBufferSize,
-		submitDelay:           options.SubmitDelay,
-		retryAttempts:         options.RetryAttempts,
-		gracefulStopScheduled: make(chan bool),
-		gracefulStopEnqueued:  make(chan bool),
-		Closer:                utils.NewCloseBroadcaster(),
+		events:          make(chan []*SubmittedEvent[T], 1),
+		submissionQueue: make(chan []*SubmittedEvent[T], 1),
+		eventsClosed:    make(chan struct{}),
+		submit:          options.Submit,
+		clock:           options.Clock,
+		submitClock:     options.SubmitClock,
+		minBatchSize:    options.MinBatchSize,
+		maxBatchSize:    options.MaxBatchSize,
+		maxBatchAge:     options.MaxBatchAge,
+		maxBufferSize:   options.MaxBufferSize,
+		submitDelay:     options.SubmitDelay,
+		retryAttempts:   options.RetryAttempts,
 	}
+
+	// lowered when Run returns
+	reporter.wg.Add(1)
```

Comment:
This is called here because `Run` itself is typically started from within a separate goroutine, right?

In theory this means that calling `UsageReporter.GracefulStop` without calling `UsageReporter.Run` first is going to hang forever or until the context is canceled.

I don't think this is a blocking issue, maybe something to consider refactoring in the future.

Reply:
>This is called here because `Run` itself is typically started from within a separate goroutine, right?

Yes, `Run` is always run from a separate goroutine and we didn't want to modify the public API (it is used in `e` too 🙄).

Reply:
As per [`(*sync.WaitGroup).Add`](https://pkg.go.dev/sync#WaitGroup.Add):
> Note that calls with a positive delta that occur when the counter is zero must happen before a Wait.

If we want to maintain the blocking/goroutine-y API of `Run`, we need to preemptively add 1 to the `WaitGroup`.

Maybe we should add a note to `GracefulStop` saying that it's illegal to call it if `Run` hasn't been ran?

Reply:
That would be good enough for now, yeah.

Reply:
Done